Serializer/Jms/Handler/ResourceHandler.php

```diff
@@ -86,16 +86,15 @@ private function doSerializeResource(Resource $resource, $depth = 0)
 
         $data['path'] = $resource->getPath();
         $data['repository_path'] = $resource->getRepositoryPath();
-        $children = $resource->listChildren();
 
         $enhancers = $this->enhancerRegistry->getEnhancers($repositoryAlias);
 
         $children = array();
-        foreach ($resource->listChildren() as $name => $resource) {
+        foreach ($resource->listChildren() as $name => $childResource) {
             $children[$name] = array();
 
-            if ($depth < 1) {
-                $children[$name] = $this->doSerializeResource($resource, $depth + 1);
+            if ($depth < 2) {
```

Comment:
`$depth < 1` is only true if `$depth` is `0`, which means `children` never contains children.

btw, I think we should make this depth configurable.